linera-core/src/client.rs

```diff
@@ -1123,12 +1128,12 @@ where
         } else {
             HashedCertificateValue::new_validated(executed_block)
         };
-        // Collect the blobs required for execution.
+        // Collect the hashed certificate values required for execution.
         let committee = self.local_committee().await?;
         let nodes = self.validator_node_provider.make_nodes(&committee)?;
-        let blobs = self
+        let values = self
             .node_client
-            .read_or_download_blobs(nodes, block.bytecode_locations())
+            .read_or_download_hashed_certificate_values(nodes, block.bytecode_locations())
```

Comment:
I hope we're only downloading the certificate values, and then hash them.

In general, I'd omit the "hashed" part in all the identifiers: It's conceptually the same; the memoized hash is an implementation detail.

Reply:
We download the actual `HashedCertificateValue` object here.
I could rename things! My only concern is that there's a `CertificateValue` and a `HashedCertificateValue` object, so if we name this `read_or_download_certificate_values` for example, it might give the impression that we're downloading `CertificateValue` objects instead of `HashedCertificateValue` objects

Reply:
But this is how serialization is implemented:

```rust
impl Serialize for HashedCertificateValue {
    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
    where
        S: serde::Serializer,
    {
        self.value.serialize(serializer)
    }
}

impl<'a> Deserialize<'a> for HashedCertificateValue {
    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
    where
        D: Deserializer<'a>,
    {
        Ok(CertificateValue::deserialize(deserializer)?.into())
    }
}
```

We _are_ effectively downloading a `CertificateValue`, and then we just compute its hash on deserialization.

Anyway, maybe something for a discussion in the team meeting.